CDP4Composition/Reporting/ReportingDataSourceNode.cs

```diff
@@ -43,13 +44,47 @@ namespace CDP4Composition.Reporting
     internal class ReportingDataSourceNode<T> where T : ReportingDataSourceRow, new()
     {
         /// <summary>
-        /// A dictionary of all the <see cref="ReportingDataSourceColumn{T}"/>s declared
-        /// as <see cref="ReportingDataSourceRow"/> fields.
+        /// A <see cref="Dictionary{TKey,TValue}"/> of all the <see cref="ReportingDataSourceColumn{T}"/>s
+        /// declared as <see cref="ReportingDataSourceRow"/> fields.
         /// </summary>
         private static readonly Dictionary<Type, FieldInfo> RowFields = typeof(T).GetFields()
             .Where(f => f.FieldType.IsSubclassOf(typeof(ReportingDataSourceColumn<T>)))
             .ToDictionary(f => f.FieldType, f => f);
 
+        /// <summary>
+        /// A <see cref="IEnumerable{T}"/> of all the public getters on the <see cref="ReportingDataSourceRow"/>
+        /// representation.
+        /// </summary>
+        private static readonly IEnumerable<PropertyInfo> PublicGetters = typeof(T).GetProperties()
+            .Where(p => p.GetMethod?.IsPublic == true);
+
+        /// <summary>
+        /// Creates a <see cref="DataTable"/> representation based on the <see cref="ReportingDataSourceRow"/>
+        /// representation.
+        /// </summary>
+        /// <param name="categoryHierarchy">
+        /// The <see cref="CategoryHierarchy"/> based on which to construct the column definitions.
+        /// </param>
+        /// <returns>
+        /// The <see cref="DataTable"/> representation.
+        /// </returns>
+        internal static DataTable GetTable(CategoryHierarchy categoryHierarchy)
+        {
+            var table = new DataTable();
+
+            for (var hierarchy = categoryHierarchy; hierarchy != null; hierarchy = hierarchy.Child)
+            {
+                table.Columns.Add(hierarchy.Category.Name, typeof(string));
```

Comment:
I think the Name property is not the best one to use for a column name. ShortName would probably be a bit safer.